aws/aws-parallelcluster
# [Isolated Regions] Fix support for US isolated regions.

*This PR makes use of cookbook changes in https://github.com/aws/aws-parallelcluster-cookbook/pull/1800*

### Description of changes
Fix support for US isolated regions.

In particular:

1. Explicitly configure the CloudFormation endpoint and region used by cfn-bootstrap tools (cfn-hup, cfn-init, cfn-signal).
1. Add a step to configure cluster nodes when running in US isolated regions. The following conditional actions have been added to the user data for both head and compute nodes: 
    a. execute the script '/opt/parallelcluster/scripts/patch-iso-instance.sh'. 
    b. reload the user profile so that 'root' can execute the other user data action with the new/updated environment variables provided by the patching script.
1. Set default EBS volume type to 'gp2' is US isolated regions. The default volume type is reflected in: 
    a. head node root volume. 
    b. compute resource root volume. 
    c. shared EBS volumes.

### Tests
Manual tests:
1. Cluster creation (executed both in Commercial and US Isolated)
3. Cluster update with dynamic file system mounting (executed both in Commercial and US Isolated)
4. Cluster update with compute nodes min-count change (executed both in Commercial and US Isolated)

A full regression tests will be done on the authoritative pipeline once the change will be merged.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Signed-off-by: Giacomo Marciani <mgiacomo@amazon.com>
